### PR TITLE
Cohesion metric notes

### PR DESCRIPTION
**Pull request by silverCase**
#182

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#214](https://github.com/yegor256/aibolit/issues/214) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/db18fbff-9c3b-4dc7-a59a-ab9296ecbce6); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by acheshkov** [comment]
@0crat out

File: aibolit/metrics/coh/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Cohesion with LCOM4
+For `class A` with methods (functions: `func1`, `func2`, `func3`) and with attributes (`attr1`, `attr2`, `attr3`) we can build a graph `G1`:
+
+![asd](https://github.com/silverCase/aibolit/blob/metrics/aibolit/metrics/coh/G1.png)
+
+To measure the cohesion of class we will use `LCOM4` metric.\
+To understand `LCOM4` take a look at `LCOM3`:\
+`LCOM3` - calculates the number of connected component in the method graph. The methods in the graph are connected,
```
Comment (acheshkov) [inline aibolit/metrics/coh/readme.md:8]: > the number of connected component

what is the definition of the component?

Comment (acheshkov) [inline aibolit/metrics/coh/readme.md:8]: > The methods in the graph

what does it mean?

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/aibolit/issues/214#issuecomment-614151185))

@acheshkov GitHub issue is still open, won't close

**Comment by acheshkov** [comment]
@silverCase 

1. Please use default terms when you talk about graph
2. Why two different graphs on the pictures have the same name `G1`?

File: aibolit/metrics/coh/readme.md (hunk 1 above)
Comment (silverCase) [reply aibolit/metrics/coh/readme.md:8]: In graph theory, a component, sometimes called a connected component, of an undirected graph is a subgraph in which any two vertices are connected to each other by paths, and which is connected to no additional vertices in the supergraph.
From [Wikipedia](https://en.wikipedia.org/wiki/Component_(graph_theory))

Comment (silverCase) [reply aibolit/metrics/coh/readme.md:8]: The vertices that represent methods in the graph

File: aibolit/metrics/coh/readme.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Cohesion with LCOM4
+For `class A` with methods (functions: `func1`, `func2`, `func3`) and with attributes (`attr1`, `attr2`, `attr3`) we can build a graph `G1`:
+
+![asd](https://github.com/silverCase/aibolit/blob/metrics/aibolit/metrics/coh/G1.png)
```
Comment (fanifieiev) [inline aibolit/metrics/coh/readme.md:4]: @silverCase Why is the link to your personal branch ?

File: aibolit/metrics/coh/readme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+To measure the cohesion of class we will use `LCOM4` metric.\
+To understand `LCOM4` take a look at `LCOM3`:\
+`LCOM3` - calculates the number of connected component in the method graph. The methods in the graph are connected,
+if methods have access to the same attribute.\
+![aasd](https://github.com/silverCase/aibolit/blob/metrics/aibolit/metrics/coh/G2.png)
```
Comment (fanifieiev) [inline aibolit/metrics/coh/readme.md:10]: @silverCase Why is the link to your personal branch ?

**Comment by 0crat** [comment]
@fanifieiev[/z](https://www.0crat.com/u/fanifieiev) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
Job audit: performer @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)[/z](https://www.0crat.com/u/fanifieiev) didn't make CR comments

**Comment by 0crat** [comment]
Pull request [#214](https://github.com/yegor256/aibolit/issues/214) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Pull request closed without merging**